test/e2e/specs/all-features/recordset/saved-query.spec.ts

```diff
@@ -0,0 +1,254 @@
+
+// utils
+import { getCatalogID, getEntityRow, updateCatalogAnnotation } from '@isrd-isi-edu/chaise/test/e2e/utils/catalog-utils';
+import { testShareCiteModal } from '@isrd-isi-edu/chaise/test/e2e/utils/record-utils';
+import { clearInput } from '@isrd-isi-edu/chaise/test/e2e/utils/recordedit-utils';
+
+const testParams = {
+  table_name: 'main',
+  minInputClass: 'range-min',
+  minInputClearClass: 'min-clear',
+  maxInputClass: 'range-max',
+  maxInputClearClass: 'max-clear',
+  firstSavedQueryName: 'main with int_col ( 11 to 22)',
+  secondSavedQueryName: 'main with int_col ( 11 to 22); one',
+  key: {
+    name: 'id',
+    operator: '=',
+    value: '1'
+  }
+};
+
+test.describe('View recordset page and form a query,', () => {
+
+  test.beforeAll(async ({}, testInfo) => {
+    const catalogId = getCatalogID(testInfo.project.name);
+
+    await updateCatalogAnnotation(catalogId, {
+      'tag:isrd.isi.edu,2019:chaise-config': {
+        savedQueryConfig: {
+          defaultName: {
+            totalTextLimit: 80,
+          },
+          storageTable: {
+            catalog: catalogId,
+            schema: 'saved_query',
+            table: 'saved_query',
+            columnNameMapping: {
+              queryId: 'query_id',
+              catalog: 'catalog',
+              schemaName: 'schema_name',
+              tableName: 'table_name',
+              userId: 'user_id',
+              queryName: 'name',
+              description: 'description',
+              facets: 'facets',
+              encodedFacets: 'encoded_facets',
+              lastExecutionTime: 'last_execution_time'
+            }
+          }
+        }
+      },
+      'tag:misd.isi.edu,2015:display': {
+        show_saved_query: true
+      }
+    });
+  });
+
+  test('For table ' + testParams.table_name + ',', async ({ page, baseURL }, testInfo) => {
+
+    await test.step('should load recordset page', async() => {
+      const PAGE_URL = `/recordset/#${getCatalogID(testInfo.project.name)}/saved_query:${testParams.table_name}`;
+
+      await page.goto(`${baseURL}${PAGE_URL}`);
+      await RecordsetLocators.waitForRecordsetPageReady(page);
+    });
+    
+    await test.step('should show a saved query dropdown', async () => {
+      const sqDropdown = RecordsetLocators.getSavedQueryDropdown(page);
+      const savedQueryMenuItems = RecordsetLocators.getSavedQueryOptions(page);
+
+      await expect.soft(sqDropdown).toBeVisible();
+
+      await sqDropdown.click();
+      await expect.soft(savedQueryMenuItems).toHaveCount(2);
+
+      // close the dropdown
+      await sqDropdown.click();
+    });
+
+    await test.step('apply filter in int_col facet and open save query form', async () => {
+      const facetIdx = 1;
+      const facet = RecordsetLocators.getFacetById(page, facetIdx);
+
+      // inputs
+      const minInput = RecordsetLocators.getRangeInput(facet, testParams.minInputClass);
```

Comment:
I think if you do the change that I mentioned in the locators, it also solves the redundancy that we have here. That way you're just calling one function and then you can use its props.